### DISCUSSION TITLE
Hey all looking to get involved in my First Open Source Project

### DISCUSSION THREAD

**Discussion by dsfai2020** [TodoLodo2089/cod-python-api#27] [General]
I like Python and Call of Duty DMZ.  I know they recently put a hold on DMZ updates but I'd like to use the API wrapper to tinker around with something that could provide DMZ players with some extra analytics.  Is this the right place?

DMZ runs on the Warzone 2.0 engine  (independent of WZ3 and the MW3 game that just came out).  The servers are promised to remain live (just not getting new content at the moment).  The battle-pass has been unlinked from Warzone entirely.  I feel like it's a safe  stable place to tinker with.

It's still live and has people actively playing.  If this API wrapper worked on WZ2 I'm hoping that I could work here.  I will try to test it out sometime soon unless the Holidays overwhelm me lol.  Happy Holidays all.

**Comment by aaron-fredrick** [comment]
Hey, sorry for our delay in responding to you.

I would like to apologize since our developers have been a bit inactive on this development with the overwhelming amount of work with other projects we have and COD franchise drastically increasing and state changes with their APIs, but we do like to make some improvements to this project once we get the time.

If you were saying you would like to contribute to our project, feel free to make a pull request and we will check your pull request and merge accordingly, if we get more contributors, we may open feature and dev branches to work along with

And very late mention but hope you had a great holiday and wishing a happy new year as well!

**Reply by dsfai2020** [reply]
Thank you so much and no apology needed!  Thanks for taking the time to respond to me.  I would love to contribute to this project and will explore it to the best of my knowledge.  My Michigan Wolverines won (I'm a fan not an Alumni) so I am having a VERY Happy New year! haha.  Thanks again for the response and explanation.

**Reply by aaron-fredrick** [reply]
hey, hope you are doing well,

Sure we can discuss on this and you are free to contribute

happy coding

**Reply by dsfai2020** [reply]
Thanks a lot! I am doing well and hope you are too.  I appreciate the encouragement!  I'm open to helping out anywhere I can.  When it comes down to the API I would really love it if we can give players from the DMZ or MW3 WZ (resurgence/ranked, big map etc...) some extra analytics.  Personally, I've always been more of a strategical player than an actual beast or 'sweaty'! lol.  I'm pretty slow paced and look at my map a lot to decide where we want to move or rotate before circle begins to close.  Same way in DMZ.  But my play style isn't really considered normal in my circle lol.  I wanted a way to help showcase players in the community that don't have high kills, but still make a difference on a team.  Little things like pings, communication, buy station buy backs, picking up fallen players, buying UAV's these all matter.  But many people only see Kills and KDR as a way of validating others' contributions to the team.  I played some ranked resurgence and was able to reach diamond 1 just by playing for placement and being selective about my gun fights.  So I guess what I am saying is that I am looking for a way to help contribute to the API so that players can gather more feedback and verification of their contributions.  (One thing we all love is that at the end of the games in -warzone mw3, it shows what playstyle the player had).  You might finish as a 'Sniper', 'Pacifist', 'Commando' or many other things.  This has helped a lot especially when a teammate has 0 kills but others can see that they helped through other ways.  I want to build on this concept but I also don't want to derail the project.  What is the best way that a newcomer like myself could get their hands dirty without actually destroying the whole thing lol.  :)  Also do you guys use discord to communicate or is everything done through Github?


(Sorry for the really long reply.)

**Reply by aaron-fredrick** [reply]
Hey, I am very interested in your idea that you have mentioned giving a more breakdown of a feedback rather than the basic raw data, I myself have moved towards games like csgo (cs2 now) and there we have something similar where we can get the contribution we do as a team member, and from the key points you mentioned yes it is a great idea to show players/users a break down of their utility, etc and the impact of those when it comes down to a team based comp match ranging from SnD to battle royal

Although this may come as some unfortunate news but before, let me breakdown how we use to do our development. We initially started this as a sort of child project based on the node,js api wrapper that used to exist in parallel to this, and the community use to research new endpoints and we (the devs and some community players) used to play matches and test the outcome and figuring out the codes for weapons, killstreaks, utilities and so on, then we test it and publish it in version releases, where we set the workflow to push to pypi, but as mentioned before with the overwhelming releases of new stuff by COD franchise and their revamps we got out of touch because their endpoints might have changed, results of those endpoints might have changed and so on
So to reinitiate this development we need to run test on the current version figure out if the headers are the same and then some more test related endpoints and response data, then we need to find the endpoints for the new games and so on

**Reply by aaron-fredrick** [reply]
@Engineer152

**Reply by Engineer152** [reply]
This definitely sounds like an interesting idea. Unfortunately there have been some drastic updates to the endpoints from Activision prior to the release of WZ2. All endpoints are now hidden behind stronger authentication walls only accessible through the in game UI. I have been keeping up with the updates to the endpoints, but have not seen any changes recently.

**Reply by dsfai2020** [reply]
@TodoLodo That is awesome about csgo I didn't know they track stuff like that.   Thank you! CS has pioneered so much in what we see today in competitive shooter games that I can't help but see it's signature in so many things so it is nice to know that csgo has some of those features that you mentioned.  That's the word I was looking for 'Utility' haha! yes that is for sure what I'd like to highlight.  Also I think it's really interesting the who methodology behind the API endpoints are gathered through results in-game gameplay data.  Searching for headers to build endpoints sounds like fun!  Node.js + Python is nice!  Those are the 2 languages that I've focused most of my learning on over the years (always learning still lol).  Python being my first and JS (mainly with React) being my second.  I can see how crazy it could be to keep up with this project with the ever changing builds and security measures continuously flowing in with weekly/daily updates.  Especially if it many of the foundational points like the headers you mentioned were gathered on the older versions of the game.  Still you've given me a lot valuable insight as to how this works and comes together.  I've seen some sites track player rankings and I'm wondering if the methods they use are legal or if web scraping with something like Beautiful soup is allowed to 'mimic' a log in through the browser and gather leaderboard data.  Please forgive me if I am over stepping here but is there a way to access some of the headers from player stats like KDR, # of Top 5's, Live 250 Leaderboards in Ranked, Avg Score etc...  The reason I am asking is because there might be a way to construct some extra stats off of those already existing ones.  (Those extra stats being the utility of players - it doesn't have to be directly post game but could be something that takes a look at players play in sections.  Last 10 games.  Games after 9PM.  Avg KDR on Wednesday.  Score/to WIn ratio...a lot of random stats that highlight utility as well as others.)  I did something like this before using pandas (python) and 'feature engineering' by combing some pre-existing columns into new ones to help some of my teammates see stats using a discord bot.  But a lot of the input was manual and I was never really able to connect to an API or gather any headers.  If something should arise here that needs any help with anything I am ready and willing.  I am a bit of a stat rat lol and mix and match all types of categories to find new ones.  Its the Trading Card Game - Deck Builder in me that never left haha.

**Reply by dsfai2020** [reply]
@Engineer152 Hello Engineer152 and Thanks!  Ah I see the Authenticator walls have increased and the headers are Only accessible through IN game ui.    It seems the bright side is that security has been added to at least help the prevention of cheating.  The barrier has been sealed and I can accept that.  I'll definitely stay tuned and keep an ear open for anything that I find.  Thank you for your insight.

**Reply by aaron-fredrick** [reply]
packet sniffing time i suppose :)
